src/rpc/mod.rs

```diff
@@ -0,0 +1,106 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use crate::config::Config;
+
+use eyre::Result;
+
+use ethers::{
+    providers::{Middleware, Provider},
+    types::{Block, BlockId, H256},
+    utils::keccak256,
+};
+
+use jsonrpsee::{
+    core::{async_trait, Error},
+    proc_macros::rpc,
+    server::ServerBuilder,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[rpc(server, client, namespace = "optimism")]
+pub trait Rpc {
+    #[method(name = "outputAtBlock")]
+    async fn output_at_block(&self, block_number: u64) -> Result<OutputRootResponse, Error>;
+}
+
+#[derive(Debug)]
+pub struct RpcServerImpl {
+    config: Arc<Config>,
+}
+
+#[async_trait]
+impl RpcServer for RpcServerImpl {
+    async fn output_at_block(&self, block_number: u64) -> Result<OutputRootResponse, Error> {
+        let l2_provider =
+            Provider::try_from(self.config.l2_rpc_url.clone()).expect("invalid L2 RPC url");
+
+        let block = l2_provider.get_block(block_number).await.unwrap().unwrap();
+
+        let state_root = block.state_root;
+        let block_hash = block.hash.unwrap();
```

Comment:
handle error

Reply:
I am handling this one using `match` pattern. Does that makes sense @ncitron ?

Reply:
Using match works. However a cleaner pattern to handle this would be to use the `ok_or` method of optionals.

You should be able to do something like `block.hash.ok_or(Error::Custom("unable to get block".to_string()))?`